### ISSUE TITLE
Breaking Changes in v3?

### ISSUE THREAD

**Issue by karlhorky** [wildbit/postmark.js#101]
Hi @ibalosh, thanks for your work on this Postmark Node.js lib!

I saw that version `3.0.0` dropped recently, which would indicate breaking changes, according to [semver](https://semver.org/).

I tried reading the release notes below from [the changelog](https://github.com/wildbit/postmark.js/blob/master/CHANGELOG.md), but I couldn't figure out which of these is a breaking change and how:

```
- Updated message response for sending
- Decoupled and refactored http client part of the library
```

I also looked in the Wiki for any kind of upgrade guide (eg. similar to [the 1.x to 2.x migration guide](https://github.com/wildbit/postmark.js/wiki/Migrating-from-1.x-to-2.x-version)), but I couldn't find anything either.

What are the breaking changes in 3.0.0? And how can users upgrade?

**Comment by ibalosh** [comment]
Hi

there are no breaking changes in v3, upgrade will not affect library usage in any way. That is why there are no additional notes about it.

However, v3 decouples http client completely from the postmark client comparex to v2.x so we can easily switch from axios if needed. Since this affects the core of the library and every operation, we upgraded version to v3. Its not 100% semver compatible but the change is significant and therefore we decided to call it v3.
